src/sparse/KokkosSparse_BlockCrsMatrix.hpp

```diff
@@ -0,0 +1,940 @@
+            }
+            else {
+              local_row_values[lcol] = vals[ offset_into_vals + lrow*block_size + lcol];
+            }
+          }
+        }
+        ++numValid;
+      }
+    } // end for ncol
+    return numValid;
+  }
+
+  //! Attempt to assign the input matrix to \c *this.
+  // Are the CUDA sparse handles needed to be copied here??
+  template<typename aScalarType, typename aOrdinalType, class aDevice, class aMemoryTraits,typename aSizeType>
+  BlockCrsMatrix&
+  operator= (const BlockCrsMatrix<aScalarType, aOrdinalType, aDevice, aMemoryTraits, aSizeType>& mtx)
+  {
+    numCols_ = mtx.numCols ();
+    blockDim_ = mtx.blockDim ();
+    graph = mtx.graph;
+    values = mtx.values;
+    dev_config = mtx.dev_config;
+    return *this;
+  }
+
+  //! The number of rows in the sparse matrix.
+  KOKKOS_INLINE_FUNCTION ordinal_type numRows () const {
+    return graph.numRows ();
+  }
+
+  //! The number of columns in the sparse matrix.
+  KOKKOS_INLINE_FUNCTION ordinal_type numCols () const {
+    return numCols_;
+  }
+
+  //! The block dimension in the sparse block matrix.
+  KOKKOS_INLINE_FUNCTION ordinal_type blockDim () const {
+    return blockDim_ ;
+  }
+
+  //! The number of stored entries in the sparse matrix.
+  KOKKOS_INLINE_FUNCTION size_type nnz () const {
+    return graph.entries.extent (0);
+  }
+
+  friend struct SparseBlockRowView<BlockCrsMatrix>;
+  //friend struct KokkosSparse::SparseRowView<BlockCrsMatrix>;
+
+  /// \brief Return a SparseBlockRowView of block-row i of the matrix.
+  ///
+  /// If row i does not belong to the matrix, return an empty view.
+  ///
+  /// The returned object \c view implements the following interface:
+  /// <ul>
+  /// <li> \c view.length is the number of entries (i.e. blocks) 
+  ///      in the block row </li>
+  /// <li> \c view.local_row_in_block_row(K, i) returns a nonconst pointer
+  ///      to the values of the ith local row in the k-th block of the block-row </li>
+  /// <li> \c view.full_row_in_block_row(i) returns a nonconst pointer
+  ///      to the values of the ith local row of the block-row </li>
+  /// <li> \c view.local_block_value(K, i, j) returns a nonconst reference
+  ///      to the value in the ith local row and jth local col 
+  ///      of the k-th block of the block-row </li>
+  /// <li> \c view.block(K) returns an unmanaged 2D strided Kokkos::View
+  ///      of the values of the k-th block of the block-row </li>
+  /// </ul>
+  ///
+  /// Users should not rely on the return type of this method.  They
+  /// should instead assign to 'auto'.
+  ///
+  KOKKOS_INLINE_FUNCTION
+  SparseBlockRowView<BlockCrsMatrix> block_row (const ordinal_type i) const {
+
+    const size_type start = graph.row_map(i); // total num blocks prior to this block-row
+    const ordinal_type count = static_cast<ordinal_type> (graph.row_map(i+1) - start); // num blocks in this row
+
+    if (count == 0) {
+      return SparseBlockRowView<BlockCrsMatrix> (NULL, NULL, 1, 0);
```

Comment:
Please use `nullptr` instead of `NULL`; thanks!

Reply:
Good catch thanks! Done.